packages/main/src/DayPicker.js

```diff
@@ -637,20 +789,39 @@ class DayPicker extends UI5Element {
 
 	get _maxDate() {
 		if (this.maxDate) {
-			const jsDate = new Date(this.getFormat().parse(this.maxDate).getFullYear(), this.getFormat().parse(this.maxDate).getMonth(), this.getFormat().parse(this.maxDate).getDate());
-			const oCalDate = CalendarDate.fromTimestamp(jsDate.getTime(), this._primaryCalendarType);
-			return oCalDate.valueOf();
+			return this._getTimeStampFromString(this.maxDate);
 		}
-		return this.maxDate;
+
+		const maxDate = new CalendarDate(1, 0, 1, this._primaryCalendarType);
+		maxDate.setYear(9999);
+		maxDate.setMonth(11);
+		const tempDate = new CalendarDate(maxDate, this._primaryCalendarType);
+		tempDate.setDate(1);
+		tempDate.setMonth(tempDate.getMonth() + 1, 0);
+		maxDate.setDate(tempDate.getDate());// 31st for Gregorian Calendar
+		return maxDate.valueOf();
 	}
 
 	get _minDate() {
 		if (this.minDate) {
-			const jsDate = new Date(this.getFormat().parse(this.minDate).getFullYear(), this.getFormat().parse(this.minDate).getMonth(), this.getFormat().parse(this.minDate).getDate());
-			const oCalDate = CalendarDate.fromTimestamp(jsDate.getTime(), this._primaryCalendarType);
+			return this._getTimeStampFromString(this.minDate);
+		}
+
+		const minDate = new CalendarDate(1, 0, 1, this._primaryCalendarType);
```

Comment:
Same here.

Reply:
Same workaround. This is also done in the SAPUI5 project as part of a bug fix.